modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1769,7 +1770,95 @@ def sort_by(
         PandasDataframe
             A new PandasDataframe sorted into lexicographical order by the specified column(s).
         """
-        pass
+        columns = columns if isinstance(columns, list) else [columns]
+        axis = Axis(axis)
+        if axis == Axis.ROW_WISE and not (
```

Comment:
could you add a comment explaining what this does? What differentiates the `if` statement vs the `elif` statement. I'm guessing we just don't need to do the rebalancing steps in the `if` case right?

Reply:
The elif is a fast path for if this sort is a just a sort by index. If that's the case, we can just create column axis partitions and run a `sort_index` in parallel across all of them - we don't need to do any shuffling etc. I'll add that as a comment.